### ISSUE TITLE
[Bug]: Nvidia B300 perf gain only 8 % on comparing with GB200.

### ISSUE THREAD

**Issue by akasshdeep** [NVIDIA/TensorRT-LLM#12651]
### System Info

-CPU - x86
-GPU - B300
-Tensorrt-llm - 1.2.0 
bduser@sclab133smc-b300:/mnt/nvme1n1$ nvidia-smi 
Tue Mar 31 23:09:31 2026       
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 580.126.20             Driver Version: 580.126.20     CUDA Version: 13.0     |
+-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA B300 SXM6 AC            On  |   00000000:10:00.0 Off |                    0 |
| N/A   62C    P0           1056W / 1100W |  247892MiB / 275040MiB |     95%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+

### Information

- [ ] The official example scripts
- [x] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

I am running a stac-benchmark on gb200 and B300, model. The Model is llama 3.1-8b-fp8 with NVFP4 quantization. I am seeing perf improvement of only 8%. As per the b300 docs it says it works 1.5x better with nvfp4.  Even I found that enabling fuse_fp4_quant is even degrading the performance like 35%. Below is my engine build:

trtllm-build \
  --max_batch_size 2048 \
  --max_seq_len 10240 \
  --max_num_tokens 16384 \
  --opt_num_tokens 14000 \
  --tokens_per_block 64 \
  --kv_cache_type paged \		
  --gemm_plugin nvfp4 \	
  --fuse_fp4_quant enable \		
  --context_fmha enable \			
  --use_paged_context_fmha enable \   
  --use_fp8_context_fmha enable \   
  --remove_input_padding enable \				
  --use_fused_mlp enable \			
  --norm_quant_fusion enable \   
  --reduce_fusion enable   
 
Dataset Params: 
Total Prompts >> 21,816
Input Tokens	>> 318 – 9,269 (avg 2,596)
Max Output Tokens >> 500
Avg Effective Seq Len	~3,096

### Expected behavior

Expectation is to get atleast 40-50% better perf on B300

### actual behavior

NA

### additional notes

NA

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and checked the [documentation](https://nvidia.github.io/TensorRT-LLM/) and [examples](https://github.com/NVIDIA/TensorRT-LLM/tree/main/examples) for answers to frequently asked questions.

**Comment by akasshdeep** [comment]
Hi @nv-guomingz, my benchmark is tensorrt-llm backend. It's a stac-ai benchmark. I am sharing you my engine build config. 
{
    "version": "1.2.0",
    "pretrained_config": {
        "mlp_bias": false,
        "attn_bias": false,
        "rotary_base": 500000.0,
        "rotary_scaling": {
            "factor": 8.0,
            "low_freq_factor": 1.0,
            "high_freq_factor": 4.0,
            "original_max_position_embeddings": 8192,
            "rope_type": "llama3"
        },
        "residual_mlp": false,
        "disable_weight_only_quant_plugin": false,
        "moe": {
            "num_experts": 0,
            "shared_expert_intermediate_size": 0,
            "top_k": 0,
            "normalization_mode": null,
            "sparse_mixer_epsilon": 0.01,
            "tp_mode": 0,
            "device_limited_n_group": 0,
            "device_limited_topk_group": 0,
            "device_limited_routed_scaling_factor": 1.0
        },
        "remove_duplicated_kv_heads": false,
        "fc_after_embed": false,
        "use_input_layernorm_in_first_layer": true,
        "use_last_layernorm": true,
        "layer_idx_offset": 0,
        "embedding_multiplier": 1.0,
        "attention_multiplier": 1.0,
        "residual_multiplier": 1.0,
        "output_multiplier_scale": 1.0,
        "has_partial_lora_mask": false,
        "architecture": "LlamaForCausalLM",
        "dtype": "bfloat16",
        "vocab_size": 128256,
        "hidden_size": 4096,
        "num_hidden_layers": 32,
        "num_attention_heads": 32,
        "hidden_act": "silu",
        "logits_dtype": "float16",
        "norm_epsilon": 1e-05,
        "runtime_defaults": null,
        "position_embedding_type": "rope_gpt_neox",
        "num_key_value_heads": 8,
        "intermediate_size": 14336,
        "max_position_embeddings": 131072,
        "mapping": {
            "world_size": 1,
            "gpus_per_node": 8,
            "cp_size": 1,
            "tp_size": 1,
            "pp_size": 1,
            "moe_tp_size": 1,
            "moe_cluster_size": 1,
            "moe_ep_size": 1,
            "attn_tp_size": 1,
            "attn_cp_size": 1,
            "cp_config": {},
            "enable_attention_dp": false,
            "enable_lm_head_tp_in_adp": false
        },
        "quantization": {
            "quant_algo": "NVFP4",
            "kv_cache_quant_algo": "FP8",
            "group_size": 16,
            "smoothquant_val": 0.5,
            "clamp_val": null,
            "use_meta_recipe": false,
            "has_zero_point": false,
            "pre_quant_scale": false,
            "exclude_modules": [
                "lm_head"
            ],
            "mamba_ssm_cache_dtype": null
        },
        "use_parallel_embedding": true,
        "embedding_sharding_dim": 0,
        "head_size": 128,
        "qk_layernorm": false,
        "rotary_embedding_dim": 128,
        "producer": {
            "name": "modelopt",
            "version": "0.37.0"
        },
        "share_embedding_table": false,
        "bias": false,
        "rotary_pct": 1.0,
        "rank": 0,
        "decoder": "llama",
        "rmsnorm": true,
        "lm_head_bias": false,
        "tie_word_embeddings": false,
        "model_type": "llama"
    },
    "build_config": {
        "max_input_len": 1024,
        "max_seq_len": 7177,
        "opt_batch_size": 8,
        "max_batch_size": 1024,
        "max_beam_width": 1,
        "max_num_tokens": 13312,
        "opt_num_tokens": null,
        "max_prompt_embedding_table_size": 0,
        "kv_cache_type": "paged",
        "gather_context_logits": false,
        "gather_generation_logits": false,
        "strongly_typed": true,
        "force_num_profiles": null,
        "profiling_verbosity": "layer_names_only",
        "enable_debug_output": false,
        "max_draft_len": 0,
        "speculative_decoding_mode": 1,
        "use_refit": false,
        "input_timing_cache": null,
        "output_timing_cache": "model.cache",
        "lora_config": {
            "lora_dir": [],
            "lora_ckpt_source": "hf",
            "max_lora_rank": 64,
            "lora_target_modules": [],
            "trtllm_modules_to_hf_modules": {},
            "max_loras": null,
            "max_cpu_loras": null,
            "swap_gate_up_proj_lora_b_weight": true
        },
        "weight_sparsity": false,
        "weight_streaming": false,
        "plugin_config": {
            "dtype": "bfloat16",
            "bert_attention_plugin": "auto",
            "gpt_attention_plugin": "auto",
            "gemm_plugin": "nvfp4",
            "gemm_swiglu_plugin": null,
            "fp8_rowwise_gemm_plugin": null,
            "qserve_gemm_plugin": null,
            "identity_plugin": null,
            "nccl_plugin": null,
            "lora_plugin": null,
            "dora_plugin": false,
            "weight_only_groupwise_quant_matmul_plugin": null,
            "weight_only_quant_matmul_plugin": null,
            "smooth_quant_plugins": true,
            "smooth_quant_gemm_plugin": null,
            "layernorm_quantization_plugin": null,
            "rmsnorm_quantization_plugin": null,
            "quantize_per_token_plugin": false,
            "quantize_tensor_plugin": false,
            "moe_plugin": "auto",
            "mamba_conv1d_plugin": "auto",
            "low_latency_gemm_plugin": null,
            "low_latency_gemm_swiglu_plugin": null,
            "gemm_allreduce_plugin": null,
            "context_fmha": true,
            "bert_context_fmha_fp32_acc": false,
            "paged_kv_cache": true,
            "remove_input_padding": true,
            "norm_quant_fusion": false,
            "reduce_fusion": false,
            "user_buffer": false,
            "tokens_per_block": 32,
            "use_paged_context_fmha": true,
            "use_fp8_context_fmha": true,
            "fuse_fp4_quant": false,
            "multiple_profiles": true,
            "paged_state": false,
            "streamingllm": false,
            "manage_weights": false,
            "use_fused_mlp": true,
            "pp_reduce_scatter": false
        },
        "use_strip_plan": false,
        "max_encoder_input_len": 1024,
        "monitor_memory": false,
        "use_mrope": false
    }
}

**Comment by karljang** [comment]
Apologies for the delayed response. The TensorRT backend is planned for deprecation as mentioned in this PR: https://github.com/NVIDIA/TensorRT-LLM/pull/11723. In the meantime, we’re focusing our investment on developing the PyTorch workflow. Could you please try the PyTorch workflow instead.
